src/observers/link_prefetch_observer.js

```diff
@@ -145,50 +139,52 @@ export class LinkPrefetchObserver {
   #isPrefetchable(link) {
     const href = link.getAttribute("href")
 
-    if (!href || href.startsWith("#") || link.getAttribute("data-turbo") === "false" || link.getAttribute("data-turbo-prefetch") === "false") {
-      return false
-    }
+    if (!href) return false
 
-    const event = dispatch("turbo:before-prefetch", {
-      target: link,
-      cancelable: true
-    })
+    if (unfetchableLink(link)) return false
+    if (linkToTheSamePage(link)) return false
```

Comment:
As we discussed IRL, I'm not sure we should always exclude the current page. Sometimes clicking navigation to the same page is a valid way of getting latest content on a page, and in that case prefetching seems valid.

Perhaps there are edge cases to think about when fetching them, though. In any case, probably fine to leave for another PR, just wanted to note is since we're touching this part.

Reply:
You could probably argue that if the current page is prefetched it might as well morph it to the latest version. Though there are probably edge cases as you say.